packages/cli/src/zosfiles/compare/CompareBaseHelper.ts

```diff
@@ -0,0 +1,201 @@
+        this.file2Options.record = commandParameters.arguments.record2;
+        this.file1Options.volumeSerial = commandParameters.arguments.volumeSerial;
+        this.file2Options.volumeSerial = commandParameters.arguments.volumeSerial2;
+
+        if (this.file2Options?.binary == null) {
+            this.file2Options.binary = this.file1Options.binary;
+        }
+        if (this.file2Options.encoding == null) {
+            this.file2Options.encoding = this.file1Options.encoding;
+        }
+        if (this.file2Options.record == null) {
+            this.file2Options.record = this.file1Options.record;
+        }
+
+        this.seqnum = commandParameters.arguments.seqnum;
+        this.browserView = commandParameters.arguments.browserView;
+        this.contextLines = commandParameters.arguments.contextLines;
+        this.responseTimeout = commandParameters.arguments.responseTimeout;
+        this.file1Options.responseTimeout = this.responseTimeout;
+        this.file2Options.responseTimeout = this.responseTimeout;
+
+        // Clear undefined properties
+        this.file1Options = JSON.parse(JSON.stringify(this.file1Options));
+        this.file2Options = JSON.parse(JSON.stringify(this.file2Options));
+    }
+
+    /**
+     * Parse the spool description and split them into individual properties
+     * @param spoolDescription Colon-separated (:) spool descriptor
+     * @returns Object containing the name, job id, and spool id
+     */
+    public prepareSpoolDescriptor(spoolDescription: string): {jobName: string, jobId: string, spoolId: number} {
+        const descriptionSeperator: string = ":";
+        const spoolDescArr = spoolDescription.split(descriptionSeperator);
+        const jobName: string = spoolDescArr[0];
+        const jobId: string = spoolDescArr[1];
+        const spoolId: number = Number(spoolDescArr[2]);
+        return { jobName, jobId, spoolId };
+    }
+
+    /**
+     * Helper function that compare-related handlers will use to get the contents of a local file
+     * @param filePath Path to the file to compate against
+     * @returns Buffer with the contents of the file
+     */
+    public prepareLocalFile(filePath: string): Buffer {
+        const localFile = path.isAbsolute(filePath) ? filePath : path.resolve(filePath);
+        const localFileHandle = fs.openSync(localFile, 'r');
+        let lfContentBuf: Buffer;
+        try {
+            // check if the path given is of a file or not
+            try {
+                if(!fs.fstatSync(localFileHandle).isFile()){
+                    throw new ImperativeError({
+                        msg: 'Path given is not of a file, do recheck your path again'
+                    });
+                }
+            } catch (error) {
+                if (error instanceof ImperativeError) throw error;
+                throw new ImperativeError({
+                    msg: 'Path not found. Please check the path and try again'
```

Comment:
If the path is not found, the error is not going to surface with the fstat call, it is going to occur with the openSync call. That case does not appear to be handled here.

Reply:
openSync is in the outer try so the file not found error should probably be in the outer catch statement. There does not appear to be an outer catch statement. Also, closeSync may fail in the finally block, since the file handle would be undefined.